documentation/src/docs/recovery-procedures/table-segment-recovery.md

```diff
@@ -0,0 +1,203 @@
+<!--
+Copyright Pravega Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Table Segment Recovery
+
+* [Issue](#issue)
+* [Repair Procedure](#repair-procedure)
+
+
+# Issue
+One of the issues recently seen in Pravega was some kind of data corruption happening on Attribute Index Segment. The issue manifested with the below error message popping up in logs:-
+```
+ERROR i.p.s.s.h.h.PravegaRequestProcessor - [requestId=2205414166888448] Error (Segment = '_system/_tables/completedTransactionsBatch-0', Operation = 'updateTableEntries')
+io.pravega.segmentstore.server.DataCorruptionException: BTreeIndex operation failed. Index corrupted.
+...
+Caused by: io.pravega.common.util.IllegalDataFormatException: [AttributeIndex[3-423]] Wrong footer information. RootPage Offset (3842188288) + Length (142858) exceeds Footer Offset (15007837).
+```
+The original issue filed can be found [here](https://github.com/pravega/pravega/issues/6712)
+
+The Attribute Index Segment is basically a BTree Index of the data (key/value pair) that goes in its associated main Table Segment. It is internally a B+Tree (existing on Pravega Segment), and is organized as as set of Pages written to storage (see this [blog post](https://cncf.pravega.io/blog/2019/11/21/segment-attributes) here). Each Page can be an index Page or leaf Page (holding data entries) having tree nodes. Each write of Pages to Storage ends with a footer, which is a pointer to the latest location of the root node in the tree. The above error indicates that the footer for the last write of Pages to Storage has been corrupted, as it contains inconsistent information. The issue having happened so, indicates some kind of corruption in the underlying Attribute Index Segment.
```

Comment:
Please, avoid writing super long lines of text (same as code). Because this is problematic when trying to provide feedback, as reviewers do not have the ability to associate the comment to the "line" where the problem is, as there is only one super long line. Try to respect the same line length as in code, so it helps reviewers to add comments.

Reply:
have split long lines into multiple lines.